enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -162,7 +162,9 @@ def check_kernel_id(self, kernel_id):
                 self.parent.kernel_session_manager.delete_session(kernel_id)
                 raise web.HTTPError(404, "Kernel does not exist: %s" % kernel_id)
 
-    def _refresh_kernel(self, kernel_id):
+    def _refresh_kernel(self, kernel_id) -> bool:
+        if not self.parent.availability_mode or self.parent.availability_mode == "single-instance":
```

Comment:
the thought here is, incase of `s_i` mode, the kernels are already hydrated when the EG server starts..so there is not need to check the persistence for kernel ?

Reply:
Correct. The multi-kernel manager should be aware of all active kernels in this case.